shimming-toolbox/shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -170,10 +174,14 @@ def _residuals_ps_huber(self, coef, unshimmed_vec, coil_mat, factor):
             float: Residuals for least squares optimization
         """
         residuals = unshimmed_vec + coil_mat @ coef
+        # Define delta with the 90th percentile of the absolute residuals
         if self._delta is None:
-            # self._delta = np.max(np.abs(residuals))
             self._delta = np.percentile(np.abs(residuals), 90)
-        return np.mean(pseudo_huber(self._delta, residuals)) / factor + np.abs(coef).dot(self.reg_vector)
+        # Adapt the weights based on the delta value, so that they adjust depending on the linear/quadratic behavior
+        _alpha = 1 / (1 + self._delta)
+        weights = (1 - _alpha) * np.sqrt(self.mask_coefficients) + _alpha * self.mask_coefficients
```

Comment:
No point in adding _ for function-specific parameters, we haven't been doing it
```suggestion
        alpha = 1 / (1 + self._delta)
        weights = (1 - alpha) * np.sqrt(self.mask_coefficients) + alpha * self.mask_coefficients
```

Reply:
Modifications applied